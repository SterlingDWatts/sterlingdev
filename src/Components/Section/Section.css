.Section {
  padding: 16px;
}

.Section h2 {
  font-size: 32px;
  margin: 24px 0 32px;
  font-weight: 600;
  line-height: 1;
}

.Section p {
  text-align: left;
  max-width: 912px;
  margin-left: auto;
  margin-right: auto;
}

.Section p + p {
  margin-top: 16px;
}
