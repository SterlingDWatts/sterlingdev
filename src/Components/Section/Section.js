import React from "react";
import PropTypes from "prop-types";
import "./Section.css";

function Section(props) {
  return (
    <section className={`${props.className} Section`}>{props.children}</section>
  );
}

Section.propTypes = {
  className: PropTypes.string.isRequired,
};

export default Section;
