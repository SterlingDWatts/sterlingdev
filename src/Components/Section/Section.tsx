import React from "react";
import "./Section.css";

type AppProps = { className: string; children?: any };

const Section = ({ className, children }: AppProps) => {
  return <section className={`${className} Section`}>{children}</section>;
};

export default Section;
