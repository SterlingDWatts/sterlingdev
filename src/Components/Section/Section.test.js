import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Section from "./Section";

describe("Section Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Section />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Section />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
