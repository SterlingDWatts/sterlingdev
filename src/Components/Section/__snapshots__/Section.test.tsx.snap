// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> renders the UI as expected 1`] = `
<section
  className="hero Section"
/>
`;
